Fix  go lint warnings

After running golang-ci and gofmt, the following is showed 

```bash
pkg/capacity/capacity_test.go:37:19: Error return value is not checked (errcheck)                                      
        kueue.AddToScheme(scheme)                                                                                                                                                                                                             
                         ^                  
pkg/capacity/capacity_test.go:66:23: Error return value of `cache.AddCapacity` is not checked (errcheck)               
                                        cache.AddCapacity(context.Background(), &c)                                                                                                                                                           
                                                         ^                                                                                                                                                                                    
pkg/capacity/capacity_test.go:89:26: Error return value of `cache.UpdateCapacity` is not checked (errcheck)            
                                        cache.UpdateCapacity(&c)                                                       
                                                            ^                                                                                                                                                                                 
pkg/capacity/capacity_test.go:203:19: Error return value is not checked (errcheck)
        kueue.AddToScheme(scheme)                                                                                                                                                                                                             
                         ^                                                                                                                                                                                                                    
pkg/capacity/snapshot_test.go:38:19: Error return value is not checked (errcheck)                                                                                                                                                             
        kueue.AddToScheme(scheme)                                                                                                                                                                                                             
                         ^                                                                                                                                                                                                                    
pkg/capacity/snapshot_test.go:122:20: Error return value of `cache.AddCapacity` is not checked (errcheck)              
                cache.AddCapacity(context.Background(), &cap)                                                          
                                 ^                                                                                     
pkg/queue/manager.go:61:21: Error return value of `qImpl.setProperties` is not checked (errcheck)
        qImpl.setProperties(q)                   
                           ^                                                                                           
pkg/queue/manager_test.go:394:19: Error return value of `manager.AddQueue` is not checked (errcheck)                   
                manager.AddQueue(ctx, &q)                                                                              
                                ^                          
pkg/queue/manager_test.go:452:20: Error return value of `manager.AddQueue` is not checked (errcheck)                   
                        manager.AddQueue(ctx, &q)                                                                      
                                        ^                                                                                                                                                                                                     
pkg/queue/manager_test.go:462:19: Error return value of `manager.AddQueue` is not checked (errcheck)
                manager.AddQueue(ctx, &q)                                                                              
                                ^                   
pkg/scheduler/scheduler.go:200:32: Error return value of `s.capacityCache.AssumeWorkload` is not checked (errcheck)
        s.capacityCache.AssumeWorkload(newWorkload)                                                                                                                                                                                           
                                      ^                                                                                
pkg/capacity/capacity.go:120:2: S1023: redundant `return` statement (gosimple)                                         
        return                                                                                                         
        ^                                                                                                                                                                                                                                     
pkg/capacity/snapshot_test.go:292:4: SA9003: empty branch (staticcheck)
                        if m == nil {                                                                                                                                                                                                         
                        ^
make: *** [Makefile:73: ci-lint] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -56,7 +56,7 @@ fmt: ## Run go fmt against code.
 
 .PHONY: fmt-verify
 fmt-verify:
-	@out=`$(GO_FMT) -w -l -d ./...`; \
+	@out=`$(GO_FMT) -w -l -d $$(find . -name '*.go')`; \
 	if [ -n "$$out" ]; then \
 	    echo "$$out"; \
 	    exit 1; \

--- a/pkg/capacity/capacity.go
+++ b/pkg/capacity/capacity.go
@@ -117,7 +117,6 @@ func (c *Capacity) deleteWorkload(w *kueue.QueuedWorkload) {
 	}
 	c.updateWorkloadUsage(wi, -1)
 	delete(c.Workloads, k)
-	return
 }
 
 func (c *Capacity) updateWorkloadUsage(wi *workload.Info, m int64) {

--- a/pkg/capacity/capacity_test.go
+++ b/pkg/capacity/capacity_test.go
@@ -34,7 +34,9 @@ import (
 
 func TestCacheCapacityOperations(t *testing.T) {
 	scheme := runtime.NewScheme()
-	kueue.AddToScheme(scheme)
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Fatalf("Failed adding kueue scheme: %v", err)
+	}
 	cache := NewCache(fake.NewClientBuilder().WithScheme(scheme).Build())
 	steps := []struct {
 		name           string
@@ -63,7 +65,9 @@ func TestCacheCapacityOperations(t *testing.T) {
 					},
 				}
 				for _, c := range capacities {
-					cache.AddCapacity(context.Background(), &c)
+					if err := cache.AddCapacity(context.Background(), &c); err != nil {
+						t.Fatalf("Failed adding capacity: %v", err)
+					}
 				}
 			},
 			wantCapacities: sets.NewString("a", "b", "c", "d"),
@@ -86,7 +90,9 @@ func TestCacheCapacityOperations(t *testing.T) {
 					},
 				}
 				for _, c := range capacities {
-					cache.UpdateCapacity(&c)
+					if err := cache.UpdateCapacity(&c); err != nil {
+						t.Fatalf("Failed updating capacity: %v", err)
+					}
 				}
 			},
 			wantCapacities: sets.NewString("a", "b", "c", "d"),
@@ -200,7 +206,9 @@ func TestCacheWorkloadOperations(t *testing.T) {
 		},
 	}
 	scheme := runtime.NewScheme()
-	kueue.AddToScheme(scheme)
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Fatalf("Failed adding kueue scheme: %v", err)
+	}
 	client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(
 		&kueue.QueuedWorkload{
 			ObjectMeta: metav1.ObjectMeta{Name: "a"},

--- a/pkg/capacity/snapshot_test.go
+++ b/pkg/capacity/snapshot_test.go
@@ -35,7 +35,9 @@ import (
 
 func TestSnapshot(t *testing.T) {
 	scheme := runtime.NewScheme()
-	kueue.AddToScheme(scheme)
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Errorf("Failed adding queue\n%s", err)
+	}
 	cache := NewCache(fake.NewClientBuilder().WithScheme(scheme).Build())
 	capacities := []kueue.Capacity{
 		{
@@ -119,7 +121,9 @@ func TestSnapshot(t *testing.T) {
 	for _, cap := range capacities {
 		// Purposely  do not make a copy of cap. Clones of necessary fields are
 		// done in AddCapacity.
-		cache.AddCapacity(context.Background(), &cap)
+		if err := cache.AddCapacity(context.Background(), &cap); err != nil {
+			t.Fatalf("Failed adding capacity: %v", err)
+		}
 	}
 	workloads := []kueue.QueuedWorkload{
 		{
@@ -289,10 +293,9 @@ func TestSnapshot(t *testing.T) {
 	}
 	for i, c := range wantCohorts {
 		for m := range c.members {
-			if m == nil {
-
+			if m != nil {
+				m.Cohort = &wantCohorts[i]
 			}
-			m.Cohort = &wantCohorts[i]
 		}
 	}
 	if diff := cmp.Diff(wantSnapshot, snapshot, cmpopts.IgnoreUnexported(Cohort{})); diff != "" {

--- a/pkg/queue/manager.go
+++ b/pkg/queue/manager.go
@@ -58,7 +58,9 @@ func (m *Manager) AddQueue(ctx context.Context, q *kueue.Queue) error {
 		return fmt.Errorf("queue %q already exists", q.Name)
 	}
 	qImpl := newQueue()
-	qImpl.setProperties(q)
+	if err := qImpl.setProperties(q); err != nil {
+		return fmt.Errorf("failed setting queue properties %v", err)
+	}
 	m.queues[key] = qImpl
 	// Iterate through existing workloads, as workloads corresponding to this
 	// queue might have been added earlier.

--- a/pkg/queue/manager.go
+++ b/pkg/queue/manager.go
@@ -58,9 +58,7 @@ func (m *Manager) AddQueue(ctx context.Context, q *kueue.Queue) error {
 		return fmt.Errorf("queue %q already exists", q.Name)
 	}
 	qImpl := newQueue()
-	if err := qImpl.setProperties(q); err != nil {
-		return fmt.Errorf("failed setting queue properties %v", err)
-	}
+	qImpl.setProperties(q)
 	m.queues[key] = qImpl
 	// Iterate through existing workloads, as workloads corresponding to this
 	// queue might have been added earlier.
@@ -90,7 +88,8 @@ func (m *Manager) UpdateQueue(q *kueue.Queue) error {
 	if !ok {
 		return errors.New("queue doesn't exist")
 	}
-	return qImpl.setProperties(q)
+	qImpl.setProperties(q)
+	return nil
 }
 
 func (m *Manager) DeleteQueue(q *kueue.Queue) {

--- a/pkg/queue/manager_test.go
+++ b/pkg/queue/manager_test.go
@@ -35,7 +35,9 @@ import (
 // preserved and they persist after the queue recreation.
 func TestAddQueueOrphans(t *testing.T) {
 	scheme := runtime.NewScheme()
-	kueue.AddToScheme(scheme)
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Errorf("Failed adding queue\n%s", err)
+	}
 	client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(
 		&kueue.QueuedWorkload{
 			ObjectMeta: metav1.ObjectMeta{
@@ -96,7 +98,10 @@ func TestAddQueueOrphans(t *testing.T) {
 
 func TestAddWorkload(t *testing.T) {
 	scheme := runtime.NewScheme()
-	kueue.AddToScheme(scheme)
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Errorf("Failed adding queue\n%s", err)
+
+	}
 	manager := NewManager(fake.NewClientBuilder().WithScheme(scheme).Build())
 	queues := []*kueue.Queue{
 		{ObjectMeta: metav1.ObjectMeta{Namespace: "earth", Name: "foo"}},
@@ -151,7 +156,10 @@ func TestAddWorkload(t *testing.T) {
 
 func TestRequeueWorkload(t *testing.T) {
 	scheme := runtime.NewScheme()
-	kueue.AddToScheme(scheme)
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Errorf("Failed adding queue\n%s", err)
+
+	}
 	queues := []*kueue.Queue{
 		{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
 		{ObjectMeta: metav1.ObjectMeta{Name: "bar"}},
@@ -222,7 +230,9 @@ func TestRequeueWorkload(t *testing.T) {
 
 func TestUpdateWorkload(t *testing.T) {
 	scheme := runtime.NewScheme()
-	kueue.AddToScheme(scheme)
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Errorf("Failed adding queue\n%s", err)
+	}
 	now := metav1.Now()
 	cases := map[string]struct {
 		queues         []string
@@ -343,7 +353,9 @@ func TestUpdateWorkload(t *testing.T) {
 func TestHeads(t *testing.T) {
 	ctx := context.Background()
 	scheme := runtime.NewScheme()
-	kueue.AddToScheme(scheme)
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Errorf("Failed adding queue\n%s", err)
+	}
 	now := time.Now().Truncate(time.Second)
 
 	queues := []kueue.Queue{
@@ -391,7 +403,9 @@ func TestHeads(t *testing.T) {
 	}
 	manager := NewManager(fake.NewClientBuilder().WithScheme(scheme).Build())
 	for _, q := range queues {
-		manager.AddQueue(ctx, &q)
+		if err := manager.AddQueue(ctx, &q); err != nil {
+			t.Errorf("Failed adding queue\n%s", err)
+		}
 	}
 	for _, wl := range workloads {
 		wl := wl
@@ -423,7 +437,9 @@ func TestHeads(t *testing.T) {
 func TestHeadsAsync(t *testing.T) {
 	ctx := context.Background()
 	scheme := runtime.NewScheme()
-	kueue.AddToScheme(scheme)
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Errorf("Failed adding queue\n%s", err)
+	}
 	now := time.Now().Truncate(time.Second)
 	wl := kueue.QueuedWorkload{
 		ObjectMeta: metav1.ObjectMeta{
@@ -449,7 +465,9 @@ func TestHeadsAsync(t *testing.T) {
 		client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(&wl).Build()
 		manager := NewManager(client)
 		go func() {
-			manager.AddQueue(ctx, &q)
+			if err := manager.AddQueue(ctx, &q); err != nil {
+				t.Errorf("Failed adding queue\n%s", err)
+			}
 		}()
 		heads := manager.Heads(ctx)
 		if diff := cmp.Diff(wantHeads, heads); diff != "" {
@@ -459,7 +477,9 @@ func TestHeadsAsync(t *testing.T) {
 
 	t.Run("AddWorkload", func(t *testing.T) {
 		manager := NewManager(fake.NewClientBuilder().WithScheme(scheme).Build())
-		manager.AddQueue(ctx, &q)
+		if err := manager.AddQueue(ctx, &q); err != nil {
+			t.Errorf("Failed adding queue\n%s", err)
+		}
 		go func() {
 			manager.addOrUpdateWorkload(&wl)
 		}()

--- a/pkg/queue/queue.go
+++ b/pkg/queue/queue.go
@@ -53,9 +53,8 @@ func newQueue() *Queue {
 	}
 }
 
-func (q *Queue) setProperties(apiQueue *kueue.Queue) error {
+func (q *Queue) setProperties(apiQueue *kueue.Queue) {
 	q.Capacity = string(apiQueue.Spec.Capacity)
-	return nil
 }
 
 func (q *Queue) PushIfNotPresent(info *workload.Info) bool {

--- a/pkg/scheduler/scheduler.go
+++ b/pkg/scheduler/scheduler.go
@@ -198,7 +198,12 @@ func (s *Scheduler) assign(ctx context.Context, e *entry) {
 		podSet.AssignedFlavors = e.TotalRequests[i].Flavors
 	}
 	newWorkload.Spec.AssignedCapacity = kueue.CapacityReference(e.Capacity)
-	s.capacityCache.AssumeWorkload(newWorkload)
+	if err := s.capacityCache.AssumeWorkload(newWorkload); err != nil {
+		log.Error(err, "Assuming workload failed")
+		log.V(2).Info("Requeueing")
+		s.queues.RequeueWorkload(ctx, &e.Info)
+		return
+	}
 	log.V(2).Info("Workload assumed in the cache")
 
 	go func() {

--- a/pkg/scheduler/scheduler_test.go
+++ b/pkg/scheduler/scheduler_test.go
@@ -612,7 +612,9 @@ func TestSchedule(t *testing.T) {
 			})
 			ctx := ctrl.LoggerInto(context.Background(), log)
 			scheme := runtime.NewScheme()
-			kueue.AddToScheme(scheme)
+			if err := kueue.AddToScheme(scheme); err != nil {
+				t.Fatalf("Failed adding kueue scheme: %v", err)
+			}
 			clientBuilder := fake.NewClientBuilder().WithScheme(scheme).WithLists(
 				&kueue.QueuedWorkloadList{Items: tc.workloads})
 			cl := clientBuilder.Build()

